### PR TITLE
Allow passed authentication params on init

### PR DESCRIPTION
**Pull request by kaosx5s**
### Description

Thanks for contributing this Pull Request. Make sure that you send in this Pull Request to the `master` branch of this repository, add a brief description, and tag the relevant issue(s) and PR(s) below.

- Relevant Issues :

- What Changed and Why? :
  - Adds support to pass a git token
  - Adds support to pass a git username and git password

- Type of change :
  - [X] New feature
  - [ ] Bug fix for existing feature
  - [ ] Code quality improvement
  - [x] Addition or Improvement of tests
  - [x] Addition or Improvement of documentation

- Other:
  - [X] Add unit tests
  - [X] Add documentation

### CODE REVIEW

File: README.md
Comment (coreycaverly) [inline README.md:1]: Usage is a bit confusing… 
One of the following two are required and GIT_TOKEN/TOKEN is used if all are available. 
GIT_TOKEN 
-OR- 
GIT_USERNAME & GIT_PASSWORD

File: gordian/repo.py [hunk 1]
```diff
@@ -16,7 +14,7 @@
 
 class Repo:
 
-    def __init__(self, repo_name, github_api_url=None, branch=None, github=None, files=None, semver_label=None, target_branch='master', fork=False):
+    def __init__(self, repo_name, github_api_url=None, branch=None, github=None, files=None, semver_label=None, target_branch='master', fork=False, token=None):
```
Comment (AgarFu) [inline gordian/repo.py:17]: I think we should be consistent, if we allow to pass the token in the constructor we should also allow username and password, so the user can choose between environment variables or passing the bits in the constructor

File: README.md
Comment (AgarFu) [reply README.md:1]: too many options are confusing, but Github provides all them ;-(

**Review by AgarFu — DISMISSED** [review]
let's be consistent with the options we provide, if we want to allow passing credentials via constructor parameters, let's allow all of them.

Then we can do something like:
```python
def __ init__(self, username=None, password=None, token=None):
  username = os.environment.get('GITHUB_USERNAME', username)
  password = os.environment.get('GITHUB_PASSWORD', password)
  token = os.environment.get('GITHUB_TOKEN', token)
  ...
  # Check for credentials
  # Instantiate the client
```

File: gordian/repo.py (hunk 1 above)
Comment (kaosx5s) [reply gordian/repo.py:17]: 💯 , agreed - I'll get those changes in 😄

Comment (kaosx5s) [reply gordian/repo.py]: Updated in: https://github.com/argoproj-labs/gordian/pull/41/commits/181c96669de54d95489c4914854a14f067d65ab3

File: README.md
Comment (kaosx5s) [reply README.md:1]: Updated in https://github.com/argoproj-labs/gordian/pull/41/commits/181c96669de54d95489c4914854a14f067d65ab3 , let me know if that helps clarify things 🙇

**Review by coreycaverly — APPROVED** [review]
LGTM

**Review by coreycaverly — APPROVED** [review]

**Comment by coreycaverly** [comment]
@AgarFu need you to approve changes.

**Pull request merged**